api/v1beta1/ccruntime_types.go

```diff
@@ -38,6 +38,11 @@ type CcRuntimeSpec struct {
 	RuntimeName CcRuntimeName `json:"runtimeName"`
 
 	Config CcInstallConfig `json:"config"`
+
+	// EnablePeerPods is used to transparently create pods on a remote system.
+	// +optional
+	// +kubebuilder:default:=false
+	EnablePeerPods bool `json:"enablePeerPods"`
```

Comment:
Why is this API change necessary? AFAICS users can deploy peerpods CRs separately and the peerpods controller does its job.

Reply:
I can remove it from CcRuntimeSpec  but it is used in a couple of places in ccruntime_controller.go. 
I am assuming I need some sort of test there. Perhaps it can be derived from enabling peerpods in the command line argument.

https://github.com/confidential-containers/operator/blob/fed10fdfd1405bd171bdab6992e145ba5c8ab7dd/controllers/ccruntime_controller.go#L443
https://github.com/confidential-containers/operator/blob/fed10fdfd1405bd171bdab6992e145ba5c8ab7dd/controllers/ccruntime_controller.go#L253

Reply:
PS sorry it took so long to get back to you. I was on PTO.

Reply:
> but it is used in a couple of places in ccruntime_controller.go.

It's not obvious to me why ccruntime controller needs to be aware of peerpods resources.

I'd think you can do something like:

```
resources:
- ../base
- peers-pods-config.yaml
```

in  `config/samples/ccruntime/peer-pods/kustomization.yaml`. Peer Pods controller will then reconcile resources it owns.

Reply:
I am pretty new to this stuff and I was following the sandboxed operator implementation.  If peerpods is not enabled falling into this code without a check would produce errors wouldn't it?

if r.ccRuntime.Spec.EnablePeerPods {
						r.Log.Info("Disable peerpods")
						// We are explicitly ignoring any errors in peerpodconfig removal as
						// these can be removed manually if needed and this is not in the critical path
						// of operator functionality
						_ = r.disablePeerPods()
					}
Thanks

Reply:
> falling into this code without a check would produce errors wouldn't it?

As far as I can understand, code modifications in `ccruntime_controller.go` are not needed because adding PeerPods config is same as doing:

```
apiVersion: confidentialcontainers.org/v1alpha1
kind: PeerPodConfig
metadata:
  name: peerpodconfig-coco
spec:
  cloudSecretName: "peer-pods-secrets"
  configMapName:   "peer-pods-cm",
  limit: "10"
```

as part of the kustomization like I suggested above.

Reply:
OK I can certainly try it out. Will let you know. Thanks

Reply:
I added to kustomization.yaml
 resources:
- ../base
- peerpodsconfig.yaml

And my peerpodsconfig.yaml
apiVersion: confidentialcontainers.org/v1alpha1
kind: PeerPodConfig
metadata:
  name: peerpodconfig-coco
spec:
  cloudSecretName: "peer-pods-secrets"
  configMapName:   "peer-pods-cm",
  limit: "10"

However I am getting the error
kubectl create -k config/samples/ccruntime/peer-pods/
error: accumulating resources: accumulating resources from 'peerpodsconfig.yaml': MalformedYAMLError: yaml: line 5: did not find expected key in File: peerpodsconfig.yaml

Reply:
After fixing the wayward comma in peerspodsconfig.yaml it created the peerpodconfig controller
cc-operator-controller-manager-68b5979488-g75kd   2/2     Running   0          3h11m
cc-operator-daemon-install-kmk6g                  1/1     Running   0          57s
cc-operator-pre-install-daemon-nsxjd              1/1     Running   0          59s
peerpodconfig-ctrl-caa-daemon-bppq5               1/1     Running   0          58s

So far so good!

Reply:
@caavery oops, yes sorry about my syntax errors

Reply:
It was like hiding in plain sight. Totally happy for the info! Now I have a bit of naming suffix/prefix '-peer-pod' build up:
kcli@virtlab723 operator]$  k get peerpodconfigs
NAME                           AGE
peerpodconfig-coco-peer-pods   164m